src/melee.cpp

```diff
@@ -1402,24 +1462,35 @@ bool player::block_hit(Creature *source, body_part &bp_hit, damage_instance &dam
     // but it still counts as a block even if it absorbs all the damage.
     float total_phys_block = mabuff_block_bonus();
     bool conductive_weapon = weapon.conductive();
+    bool conductive_armor = armor.conductive();
 
     // This gets us a number between:
     // str ~0 + skill 0 = 0
     // str ~20 + skill 10 + 10(unarmed skill or weapon bonus) = 40
     int block_score = 1;
-    if (can_weapon_block()) {
+    if (can_armor_block()) {
         int block_bonus = 2;
-        if (weapon.has_technique( WBLOCK_3 )) {
+        if (armor.has_technique( WBLOCK_3 )) {
             block_bonus = 10;
-        } else if (weapon.has_technique( WBLOCK_2 )) {
+        } else if (armor.has_technique( WBLOCK_2 )) {
             block_bonus = 6;
-        } else if (weapon.has_technique( WBLOCK_1 )) {
+        } else if (armor.has_technique( WBLOCK_1 )) {
             block_bonus = 4;
         }
         ///\EFFECT_STR increases attack blocking effectiveness with a weapon
 
         ///\EFFECT_MELEE increases attack blocking effectiveness with a weapon
         block_score = str_cur + block_bonus + (int)get_skill_level( skill_melee );
+    } else if (can_weapon_block()) {
```

Comment:
As in the comment above: once you have a function that returns the best item capable of blocking, you won't need separate sections for armor and weapons.